File: python-backend/app/routers/knowledge.py
import uuid

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.deps import get_current_user
from app.db.base import get_db
from app.db.models.user import User
from app.schemas.knowledge import QAEntryCreate, QAEntryOut
from app.services import knowledge_service

router = APIRouter(prefix="/knowledge", tags=["knowledge"], dependencies=[Depends(get_current_user)])


@router.get("/qa", response_model=list[QAEntryOut])
def list_qa(sector: uuid.UUID, db: Session = Depends(get_db)):
    return knowledge_service.list_entries(db, sector)


@router.post("/qa", response_model=QAEntryOut, status_code=201)
def create_qa(
    payload: QAEntryCreate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return knowledge_service.create_entry(db, current_user, payload)


@router.delete("/qa/{entry_id}", status_code=204)
def delete_qa(
    entry_id: uuid.UUID,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    knowledge_service.delete_entry(db, current_user, entry_id)

File: react-app/src/api/knowledge.ts
import { apiFetch } from "./client";
import type { QAEntry, QAEntryCreate } from "./types";

export function listEntries(sectorId: string): Promise<QAEntry[]> {
  return apiFetch<QAEntry[]>("/knowledge/qa", { params: { sector: sectorId } });
}

export function createEntry(payload: QAEntryCreate): Promise<QAEntry> {
  return apiFetch<QAEntry>("/knowledge/qa", { method: "POST", body: payload });
}

export function deleteEntry(entryId: string): Promise<void> {
  return apiFetch<void>(`/knowledge/qa/${entryId}`, { method: "DELETE" });
}
